core/common/assets/js/api/modules/command-data.js

```diff
@@ -0,0 +1,182 @@
+import CommandBase from './command-base';
+
+export default class CommandData extends CommandBase {
+	/**
+	 * Data returned from remote.
+	 *
+	 * @type {*}
+	 */
+	data;
+
+	/**
+	 * Fetch type.
+	 *
+	 * @type {DataTypes}
+	 */
+	type;
+
+	static getInstanceType() {
+		return 'CommandData';
+	}
+
+	constructor( args, commandsAPI = $e.data ) {
+		super( args, commandsAPI );
+
+		if ( this.args.options?.type ) {
+			this.type = this.args.options.type;
+		}
+	}
+
+	/**
+	 * Function getEndpointFormat().
+	 *
+	 * @returns {(null|string)}
+	 */
+	getEndpointFormat() {
+		return null;
+	}
+
+	/**
+	 * @param {DataTypes} type
+	 *
+	 * @returns {boolean|{before: (function(*=): {}), after: (function({}, *=): {})}}
+	 */
+	getApplyMethods( type = this.type ) {
+		let before, after;
+		switch ( type ) {
+			case 'create':
+				before = this.applyBeforeCreate;
+				after = this.applyAfterCreate;
+				break;
+
+			case 'delete':
+				before = this.applyBeforeDelete;
+				after = this.applyAfterDelete;
+				break;
+
+			case 'get':
+				before = this.applyBeforeGet;
+				after = this.applyAfterGet;
+				break;
+
+			case 'update':
+				before = this.applyBeforeUpdate;
+				after = this.applyAfterUpdate;
+				break;
+
+			default:
+				return false;
+		}
+
+		return { before, after };
+	}
+
+	/**
+	 * Function getRequestData().
+	 *
+	 * @returns {RequestData}
+	 */
+	getRequestData() {
+		return {
+			type: this.type,
+			args: this.args,
+			timestamp: new Date().getTime(),
+			component: this.component,
+			command: this.currentCommand,
+			endpoint: $e.data.commandToEndpoint( this.currentCommand, elementorCommon.helpers.cloneObject( this.args ), this.getEndpointFormat() ),
```

Comment:
It is used in `$e.data` since we didn't normalized yet all ways we talk with cache ( should be only via `$e.data` ), and we use sometimes we `$e.data.cache` instead of `$e.data`...